SeQuant/core/eval_expr.hpp

```diff
@@ -1,190 +1,50 @@
 #ifndef SEQUANT_EVAL_EXPR_HPP
 #define SEQUANT_EVAL_EXPR_HPP
 
+#include <SeQuant/core/expr_fwd.hpp>
 #include <SeQuant/core/tensor.hpp>
 
 namespace sequant {
 
-/**
- * Type of evaluation resulting into EvalExpr.
- */
-enum class EvalOp {
-  /** Identity type evaluation. */
-  Id,
-  /** Sum type evaluation. */
-  Sum,
-  /** Product type evaluation. */
-  Prod,
-  /** Symmetrization type evaluation. */
-  Symm,
-  /** Antisymmetrization type evaluation. */
-  Antisymm,
-};  // EvalOp
-
-/**
- * An object to be stored in the binary evaluation tree nodes.
- *
- * @author Bimal Gaudel
- * @version 03 Dec, 2020
- */
-class EvalExpr {
- public:
-  using hash_t = size_t;
+enum class EvalOp { Id, Sum, Prod };
+
+enum class ResultType { Tensor, Constant };
 
-  EvalExpr() = delete;
+class EvalExpr {
+ private:
+  EvalOp op_type_;
+  ResultType result_type_;
+  size_t hash_value_;
+  size_t id_;
+  ExprPtr expr_;
+  bool tot_;
 
-  /**
-   * Construct EvalExpr that goes into the leaf nodes
-   * of the binary evaluation tree.
-   */
-  explicit EvalExpr(const Tensor&) noexcept;
+  static size_t global_id_;
```

Comment:
Typically, class definitions are easier to understand if private members are defined after public ones ('cause most of the time one looks up a class definition, one is interested in how one can interact with the class from the outside.)